src/shared/reboot-util.c

```diff
@@ -42,6 +47,9 @@ int update_reboot_parameter_and_warn(const char *parameter, bool keep) {
                 return 0;
         }
 
+        if (!reboot_parameter_is_valid(parameter))
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid reboot parameter, ignoring.");
```

Comment:
I think "ignoring" is spurious. After all the error is checked by most of the callers.